src/routing_node.rs

```diff
@@ -232,6 +233,33 @@ impl RoutingNode {
           self.set_cache_options(preserve_cache_options);
     }
 
+    fn handle_on_connect(&mut self, connection: ::crust::Connection) {
+        match self.core.state() {
+            &::routing_core::State::Disconnected => {
```

Comment:
can you add small inline comments on the motivation behind these steps?  that would be very valuable here I think.  Thanks!